content/en/os/1.22.x/concepts/bootstrap-commands/_index.markdown

```diff
@@ -0,0 +1,128 @@
+- `systemd` will not move to the next [target](https://www.freedesktop.org/software/systemd/man/systemd.target.html) until all of the bootstrap commands start, run and exit.
+- any bootstrap command configured with `mode=once` will change to `mode=off` once complete.
+- any bootstrap command set to `essential=true` that exits with a non-zero exit code halts the boot process.
+
+### Examples
+
+{{< twocol
+    containerclass="td-max-width-on-larger-screens docs-figure"
+    rowclass="docs-figure" >}}
+    {{< twocol_inner colsplit="7" >}}
+        {{< bootstrap_commands_diagram example=1 >}}
+    {{</ twocol_inner >}}
+    {{% twocol_inner %}}
+This example shows four bootstrap commands with names (`001-cmd`,`002-cmd`,`003-cmd`, and `004-cmd`) starting.
+The start order of these commands is in lexicographical order of their names.
+The next stage of the boot does not proceed until the last bootstrap command completes.
+    {{%/ twocol_inner %}}
+{{</ twocol >}}
+
+{{< twocol
+    containerclass="td-max-width-on-larger-screens docs-figure"
+    rowclass="docs-figure" >}}
+    {{< twocol_inner  colsplit="7" >}}
+        {{< bootstrap_commands_diagram example=2 >}}
+    {{</ twocol_inner >}}
+    {{% twocol_inner %}}
+This example is the same as the previous one except bootstrap command `003-cmd` is configured with {{< ver-ref project="os" page="/api/settings/bootstrap-commands#name_essential">}}`essential=true`{{< /ver-ref >}}.
+Bootstrap command `003-cmd` exits with a non-zero exit code and consequently the boot halts. As a result, `004-cmd` isn't run at all.
+    {{%/ twocol_inner %}}
+{{</ twocol >}}
+
+{{< twocol
+    containerclass="td-max-width-on-larger-screens docs-figure"
+    rowclass="docs-figure" >}}
+    {{< twocol_inner  colsplit="7" >}}
+        {{< bootstrap_commands_diagram example=3 >}}
+    {{</ twocol_inner >}}
+    {{% twocol_inner %}}
+This example is the same as the first except bootstrap command `004-cmd` is configured with {{< ver-ref project="os" page="/api/settings/bootstrap-commands#name_mode">}}`mode=once`{{< /ver-ref >}}.
+After `004-cmd` finishes, Bottlerocket (via a `systemd` unit) updates bootstrap command `004-cmd`'s `mode` to `off`.
+    {{%/ twocol_inner %}}
+{{</ twocol >}}
+
+### Considerations
+
+As a consequence of this lifecycle, you should keep a few things in mind when using bootstrap commands:
+
+1. Bootstrap commands can only run Bottlerocket API commands (specifically commands that start with apiclient). The exception to this rule is that it can’t run `apiclient exec` commands because host containers aren’t spawned during early boot.
+2. Attempting to run `apiclient reboot` in a bootstrap command with `mode=always` will cause the node to reboot endlessly.
+3. `apiclient reboot` in a bootstrap command with `mode=once` will hold off the reboot until the completion of all bootstrap commands.
+4. Due to serial nature of bootstrap commands, they should be short-running to reduce overhead on boot times.
```

Comment:
```suggestion
1. Bootstrap commands **can only** run `apiclient` commands.
2. Bootstrap commands can’t run `apiclient exec` commands because host containers aren’t ready at this stage of the boot.
3. Running `apiclient reboot` in a bootstrap command with `mode=always` will cause the node to reboot endlessly.
4. `apiclient reboot` in a bootstrap command with `mode=once` will hold off the reboot until the completion of all bootstrap commands.
5. Due to serial nature of bootstrap commands, they should be short-running to reduce overhead on boot times.
```